File: math/0883_projection_area_of_3d_shapes/0883_projection_area_of_3d_shapes.py
from typing import List

class Solution:
    def projectionArea(self, grid: List[List[int]]) -> int:
        xy = sum(grid[i][j]>0  for i in range(len(grid)) for j in range(len(grid[0])))
        xz = sum(max(i)  for i in grid)
        yz = sum(max(i)  for i in zip(*grid))

        return xy+xz+yz



grid = [[1,2],[3,4]]
res = Solution().projectionArea(grid)
print(res)

File: math/0883_projection_area_of_3d_shapes/0883_projection_area_of_3d_shapes.go
package main

import (
	"fmt"
	"math"
)

func projectionArea(grid [][]int) int {
	rowNum, colNum := len(grid), len(grid[0])
	xy, xz, yz, rowMax, colMax := 0, 0, 0, make([]int, rowNum), make([]int, colNum)
	for row, v1 := range grid {
		for col, v2 := range v1 {
			if v2 != 0 {
				xy++
			}
			rowMax[row] = int(math.Max(float64(rowMax[row]), float64(grid[row][col])))
			colMax[col] = int(math.Max(float64(colMax[col]), float64(grid[row][col])))
		}
	}
	for _, v := range rowMax {
		xz += v
	}
	for _, v := range colMax {
		yz += v
	}
	return xy + xz + yz
}

func main() {
	grid := [][]int{{2}}
	res := projectionArea(grid)
	fmt.Println(res)
}
